### PR TITLE
Code Quality: Clean up App.Shell

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Description

I didn't know what this namespace is for. So I started working on readability refactoring first (stage 2).

> 2. Rename variables with preferred name to follow the guideline (The most likely to happen is that I rename property-backing-field from '_isHidden' to '_IsHidden')

### Motivation & Context

- Improved readability
- Updated variables to be more accurate.

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #12340
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

### CODE REVIEW

File: src/Files.App/Shell/ShellNewMenuHelper.cs [hunk 1]
```diff
@@ -79,7 +80,9 @@ private static Task<ShellNewEntry> ParseShellNewRegistryEntry(RegistryKey key, R
 				!valueNames.Contains("Command", StringComparer.OrdinalIgnoreCase) &&
 				!valueNames.Contains("ItemName", StringComparer.OrdinalIgnoreCase) &&
 				!valueNames.Contains("Data", StringComparer.OrdinalIgnoreCase))
+			{
```
Comment (hecksmosis) [inline src/Files.App/Shell/ShellNewMenuHelper.cs:83]: Why did you add these curly braces?

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+		private ShellLibraryFolders _folders;
+
+		private string _name;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ShellLibrary"/>Ex class.
+		/// </summary>
+		/// <param name="knownFolderId">The known folder identifier.</param>
+		/// <param name="readOnly">if set to <c>true</c> [read only].</param>
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:28]:
```suggestion
		/// <param name="readOnly">If set to <c>true</c> [read only].</param>
```

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+			Init(knownFolderId.GetIShellItem());
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ShellLibrary"/>Ex class.
+		/// </summary>
+		/// <param name="libraryName">Name of the library.</param>
+		/// <param name="kf">The known folder identifier.</param>
+		/// <param name="overwrite">if set to <c>true</c> [overwrite].</param>
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:42]:
```suggestion
		/// <param name="overwrite">If set to <c>true</c> [overwrite].</param>
```

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+			Init(item);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ShellLibrary"/>Ex class.
+		/// </summary>
+		/// <param name="libraryName">Name of the library.</param>
+		/// <param name="parent">The parent.</param>
+		/// <param name="overwrite">if set to <c>true</c> [overwrite].</param>
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:57]:
```suggestion
		/// <param name="overwrite">If set to <c>true</c> [overwrite].</param>
```

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+			var item = _lib.Save(parent.IShellItem, libraryName, overwrite ? Shell32.LIBRARYSAVEFLAGS.LSF_OVERRIDEEXISTING : Shell32.LIBRARYSAVEFLAGS.LSF_FAILIFTHERE);
+
+			Init(item);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ShellLibrary"/> class.
+		/// </summary>
+		/// <param name="iItem">The i item.</param>
+		/// <param name="readOnly">if set to <c>true</c> [read only].</param>
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:71]:
```suggestion
		/// <param name="readOnly">If set to <c>true</c> [read only].</param>
```

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+				return l;
+			}
+			set => _lib.SetIcon(value.ToString());
+		}
+
+		/// <summary>
+		/// Gets the name relative to the parent for the item.
+		/// </summary>
+		public override string Name { get => _name; protected set => _name = value; }
+
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:117]: Shouldn't you split this into multiple lines?

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+		/// The View Template identifier.
+		/// </value>
+		public Guid ViewTemplateId
+		{
+			get => _lib.GetFolderType();
+			set => _lib.SetFolderType(value);
+		}
+
+		public void Reload()
+		{
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:159]: This method is the only one not documented

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+			_folders = GetFilteredFolders();
+		}
+
+		/// <summary>
+		/// Commits library updates.
+		/// </summary>
+		public void Commit()
+			=> _lib.Commit();
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:167]: Can you use brackets?

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+
+		/// <summary>
+		/// Gets the set of child folders that are contained in the library.
+		/// </summary>
+		/// <param name="filter">A value that determines the folders to get.</param>
+		/// <returns>A <see cref="ShellItemArray"/> containing the child folders.</returns>
+		public ShellLibraryFolders GetFilteredFolders(LibraryFolderFilter filter = LibraryFolderFilter.AllItems)
+			=> new(_lib, _lib.GetFolders<Shell32.IShellItemArray>((Shell32.LIBRARYFOLDERFILTER)filter));
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:175]: As above

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+		/// Resolves the target location of a library folder, even if the folder has been moved or renamed.
+		/// </summary>
+		/// <param name="item">A ShellItem object that represents the library folder to locate.</param>
+		/// <param name="timeout">
+		/// The maximum time the method will attempt to locate the folder before returning. If the folder could not be located before the
+		/// specified time elapses, an error is returned.
+		/// </param>
+		/// <returns>The resulting target location.</returns>
+		public ShellItem ResolveFolder(ShellItem item, TimeSpan timeout)
+			=> Open(_lib.ResolveFolder<Shell32.IShellItem>(item.IShellItem, Convert.ToUInt32(timeout.TotalMilliseconds)));
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:187]: As above

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+		/// </param>
+		/// <param name="title">
+		/// The title for the library management dialog. To display the generic title string, set the value of this parameter to NULL.
+		/// </param>
+		/// <param name="instruction">
+		/// The help string to display below the title string in the library management dialog box. To display the generic help string, set
+		/// the value of this parameter to NULL.
+		/// </param>
+		/// <param name="allowUnindexableLocations">
+		/// if set to <c>true</c> do not display a warning dialog to the user in collisions that concern network locations that cannot be indexed.
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:203]:
```suggestion
		/// If set to <c>true</c> do not display a warning dialog to the user in collisions that concern network locations that cannot be indexed.
```

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+			GC.SuppressFinalize(this);
+		}
+
+		/// <summary>
+		/// Folders of a <see cref="ShellLibrary"/>.
+		/// </summary>
+		/// <seealso cref="ShellItemArray"/>
+		/// <seealso cref="ICollection{ShellItem}"/>
+		public class ShellLibraryFolders : ShellItemArray, ICollection<ShellItem>
+		{
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:235]: Shouldn't you move it to its own file?

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+			private Shell32.IShellLibrary lib;
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:236]:
```suggestion
			private Shell32.IShellLibrary _lib;
```

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 14]
```diff
@@ -0,0 +1,303 @@
...
+
+			/// <summary>
+			/// Initializes a new instance of the <see cref="ShellLibraryFolders"/> class.
+			/// </summary>
+			/// <param name="lib">The library.</param>
+			/// <param name="shellItemArray">The shell item array.</param>
+			internal ShellLibraryFolders(Shell32.IShellLibrary lib, Shell32.IShellItemArray shellItemArray) : base(shellItemArray)
+				=> this.lib = lib;
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:244]: Can you use brackets instead?

File: src/Files.App/Shell/ShellLibraryEx.cs [hunk 15]
```diff
@@ -0,0 +1,303 @@
...
+
+			/// <summary>
+			/// Adds the specified location.
+			/// </summary>
+			/// <param name="location">The location.</param>
+			/// <exception cref="ArgumentNullException">location</exception>
+			public void Add(ShellItem location)
+			{
+				if (location is null) throw new ArgumentNullException(nameof(location));
+				lib.AddFolder(location.IShellItem);
```
Comment (ferrariofilippo) [inline src/Files.App/Shell/ShellLibraryEx.cs:260]:
```suggestion
				if (location is null)
				        throw new ArgumentNullException(nameof(location));
				        
				lib.AddFolder(location.IShellItem);
```

**Review by ferrariofilippo — COMMENTED** [review]
Most comments are about documentation, but overall LGTM

**Comment by 0x5bfa** [comment]
Thank you! I missed that file and oc I should do split. I'm going to do that later.

File: src/Files.App/Shell/ShellNewMenuHelper.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellNewMenuHelper.cs:83]: According to the guideline, we should remove brackets only if if statement body and statement body are both one line.

**Comment by 0x5bfa** [comment]
> Also, why did you remove the versioning thing?

Minimum windows version is earlier than Files.App project minimum windows version. We should add them back when we introduce Uno or something. But for now adding only there is redundant.

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 15 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:260]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 14 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:244]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 13 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:236]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 12 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:235]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 11 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:203]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 2 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:28]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 3 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:42]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 4 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:57]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 5 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:71]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 6 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:117]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 7 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:159]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 8 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:167]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 9 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:175]: Fixed

File: src/Files.App/Shell/ShellLibraryEx.cs (hunk 10 above)
Comment (0x5bfa) [reply src/Files.App/Shell/ShellLibraryEx.cs:187]: Fixed

**Review by ferrariofilippo — APPROVED** [review]
LGTM

File: src/Files.App/Shell/ShellNewMenuHelper.cs (hunk 1 above)
Comment (hecksmosis) [reply src/Files.App/Shell/ShellNewMenuHelper.cs:83]: Ok

**Review by hecksmosis — APPROVED** [review]
LGTM
